src/riscv-double-trap.adoc

```diff
@@ -200,47 +200,47 @@ _unexpected trap_ is as follows:
 
 [NOTE]
 ====
-The consequence of this specification is that on occurrence of double trap the
-RNMI handler is not provided with information that a trap would report in the
-`mtval` and the `mtval2` registers. This information, if needed, may be obtained
+The consequence of this specification is that on occurrence of double trap, the
+RNMI handler is not provided with information that a trap reports in the
+`mtval` and the `mtval2` registers. This information, if needed, can be obtained
 by the RNMI handler by decoding the instruction at the address in `mnepc` and
 examining its source register contents.
 ====
 
 * When the Smrnmi extension is not implemented, or if the Smrnmi extension is
   implemented and `mnstatus.NMIE` is 0, the hart enters a critical-error state
-  without updating any architectural state including the `pc`. This state
+  without updating any architectural state, including the `pc`. This state
   involves ceasing execution, disabling all interrupts (including NMIs), and
   asserting a `critical-error` signal to the platform.
 
 [NOTE]
 ====
-The actions performed by the platform on assertion of a `critical-error` signal
-by a hart are platform specific. The range of possible actions include restarting
-the affected hart or restarting the entire platform among others.
+The actions performed by the platform when a hart asserts a `critical-error` signal
+are platform-specific. The range of possible actions include restarting
+the affected hart or restarting the entire platform, among others.
 ====
 
 An `MRET` instruction sets the `MDT` bit to 0.
 
 The Smdbltrp extension introduces a read-write critical-error-trigger (`cetrig`)
 field (bit 19) to the `dcsr` CSR cite:[DBG]. When `cetrig` is set to 1, a hart
 in a critical error state enters Debug Mode instead of asserting the
-critical-error signal to the platform. Upon such entry into Debug Mode, the
+critical-error signal to the platform. When the hart enters Debug Mode, the
 `cause` field of `dcsr` is set to 7, and its `extcause` field is set to 0,
 indicating a critical error triggered the Debug Mode entry. This cause has the
-highest priority among all reasons for entering Debug Mode. Resuming from Debug
-Mode following an entry from the critical error state returns the hart to the
+highest priority among all reasons for entering Debug Mode. Resuming operations from 
+Debug Mode following an entry from the critical error state returns the hart to the
 critical error state.
 
 [NOTE]
 ====
-The implication of this specification is that resuming from Debug Mode,
-following an entry due to a critical error, will either result in an immediate
+The implication of this specification is that resuming operations from Debug Mode,
```

Comment:
We can keep it as is - resuming  from Debug  Mode - is a well defined thing.

Reply:
Same with this one. The noun makes it clearer.